### PR TITLE
SNOW-1694806: Add support for rolling and expanding count with timedelta.

### PR DESCRIPTION
**Pull request by sfc-gh-mvashishtha**
Fixes SNOW-1694806

Add support for rolling().count() and expanding().count() for both Series and DataFrames.

For other rolling and expanding methods, match pandas behavior, which is to raise some kind of error.

### CODE REVIEW

**Review by sfc-gh-vbudati — APPROVED** [review]
Looks great, thanks Mahesh!

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py [hunk 1]
```diff
@@ -14134,6 +14136,12 @@ def _window_agg(
                 ErrorMessage.not_implemented(  # pragma: no cover
                     f"Window aggregation does not support the aggregation {repr_aggregate_function(agg_func, agg_kwargs)}"
                 )
+            if (
+                snowflake_agg_func.snowpark_aggregation is not count
```
Comment (sfc-gh-nkrishna) [inline src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:14140]: Can you add a comment saying that `count` does not output Timedelta columns which is why it is supported?

**Review by sfc-gh-nkrishna — APPROVED** [review]
LGTM, left one comment

File: src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py (hunk 1 above)
Comment (sfc-gh-mvashishtha) [reply src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:14140]: Done.

File: tests/integ/modin/window/test_expanding.py [hunk 2]
```diff
@@ -174,3 +186,69 @@ def test_expanding_aggregation_series_unsupported(agg_func, agg_func_kwargs):
...
+            ),
+            lambda object: getattr(object.expanding(), agg_func)(numeric_only=False),
+            expect_exception=True,
+            expect_exception_type=DataError,
+            # pandas error message is different for dataframe and series, but
+            # Snowpark pandas always uses the same message. Accept either
+            # message.
+            expect_exception_match=(
+                f"(?:{re.escape('No numeric types to aggregate')})|"
+                + f"(?:{re.escape(_TIMEDELTA_ROLLING_AGGREGATION_NOT_SUPPORTED)})"
```
Comment (sfc-gh-jjiao) [inline tests/integ/modin/window/test_expanding.py:220]: qq: do we need to re.escape those literals?

**Review by sfc-gh-jjiao — APPROVED** [review]

File: tests/integ/modin/window/test_expanding.py (hunk 2 above)
Comment (sfc-gh-mvashishtha) [reply tests/integ/modin/window/test_expanding.py:220]: We want to interpret the portions like `)|` and `(?:` as regex, not as literals within the regex, so we don't need to escape them.

**Pull request merged**